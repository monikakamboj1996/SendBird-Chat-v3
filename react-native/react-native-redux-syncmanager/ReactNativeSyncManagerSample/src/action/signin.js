
const Action = {
	Type: {
		AUTHENTICATE: 'authenticate',
		SIGNIN_SUCCESS: 'signin-success',
		SIGNIN_FAILED: 'signin-failed'
	}
};
Action.authenticate = () => {
	return {
		type: Action.Type.AUTHENTICATE
	};
};
Action.succeed = user => {
	return {
		type: Action.Type.SIGNIN_SUCCESS,
		payload: { user }
	};
};
Action.fail = err => {
	return {
		type: Action.Type.SIGNIN_FAILED,
		payload: { err }
	};
};

export default Action;