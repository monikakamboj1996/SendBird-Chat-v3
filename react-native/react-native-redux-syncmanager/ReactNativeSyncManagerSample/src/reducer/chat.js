import Action from '../action/chat';
import { findMessageIndex, getMessageIndex } from '../utils';

const _initialState = {
  channel: null,
  messages: [],
  inputMode: 'chat'
};

export const reducer = (state = _initialState, action) => {
  const newState = { ...state };
  switch (action.type) {
    case Action.Type.MESSAGE_SELECT: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < newState.messages.length; i++) {
        newState.messages[i]._selected = newState.messages[i].isIdentical(action.payload.message);
      }
      break;
    }
    case Action.Type.MESSAGE_DESELECT_ALL: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < newState.messages.length; i++) {
        newState.messages[i]._selected = false;
      }
      break;
    }
    case Action.Type.MESSAGE_ADD: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < action.payload.messages.length; i++) {
        const message = action.payload.messages[i];
        const isFailed = message.messageId === 0 && message.requestState === 'failed';
        const index = findMessageIndex(message, newState.messages, isFailed);
        if (index >= 0) {
          newState.messages.splice(index, 0, message);
          if (index > 0) {
            newState.messages[index - 1]._isPreviousMessageSentBySameUser =
              newState.messages[index - 1].sender &&
              newState.messages[index].sender &&
              newState.messages[index - 1].sender.userId === newState.messages[index].sender.userId;
          }
          if (index < newState.messages.length - 1) {
            newState.messages[index]._isPreviousMessageSentBySameUser =
              newState.messages[index].sender &&
              newState.messages[index + 1].sender &&
              newState.messages[index].sender.userId === newState.messages[index + 1].sender.userId;
          }
        }
      }
      break;
    }
    case Action.Type.MESSAGE_UPDATE: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < action.payload.messages.length; i++) {
        const message = action.payload.messages[i];
        const index = getMessageIndex(message, newState.messages);
        if (index >= 0) {
          message._isPreviousMessageSentBySameUser = newState.messages[index]._isPreviousMessageSentBySameUser;
          newState.messages[index] = message;
          if (action.payload.isTransformFailedToManual) {
            newState.messages[index]._isManualResend = true;
          }
        }
      }
      break;
    }
    case Action.Type.MESSAGE_REMOVE: {
      newState.messages = [...newState.messages];
      for (let i = 0; i < action.payload.messages.length; i++) {
        const message = action.payload.messages[i];
        const index = getMessageIndex(message, newState.messages);
        if (index >= 0) {
          newState.messages.splice(index, 1);
          if (index < newState.messages.length - 1) {
            newState.messages[index]._isPreviousMessageSentBySameUser =
              newState.messages[index + 1].sender &&
              newState.messages[index].sender &&
              newState.messages[index + 1].sender.userId === newState.messages[index].sender.userId;
          }
        }
      }
      break;
    }
    case Action.Type.MESSAGE_CLEAR: {
      newState.messages = [];
      break;
    }
  }
  return newState;
};
