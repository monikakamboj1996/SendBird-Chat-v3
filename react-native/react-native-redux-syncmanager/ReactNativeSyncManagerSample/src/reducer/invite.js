import Action from "../action/invite";

const _initialState = {
	users: [],
	err: null
};

export const reducer = (state = _initialState, action) => {
	const newState = { ...state };
	switch(action.type) {
		case Action.Type.USER_FETCH: {
			newState.users = newState.users.concat(action.payload.users);
			break;
		}
	}
	return newState;
};