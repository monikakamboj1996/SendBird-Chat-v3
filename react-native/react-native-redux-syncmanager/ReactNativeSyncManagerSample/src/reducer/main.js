import Action from '../action/main';
import { findChannelIndex, getChannelIndex } from '../utils';

const _initialState = {
  channels: []
};

export const reducer = (state = _initialState, action) => {
  const newState = { ...state };
  switch (action.type) {
    case Action.Type.CHANNEL_ADD: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = findChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels.splice(index, 0, channel);
        }
      }
      break;
    }
    case Action.Type.CHANNEL_UPDATE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = getChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels[index] = channel;
        }
      }
      break;
    }
    case Action.Type.CHANNEL_REMOVE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = getChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels.splice(index, 1);
        }
      }
      break;
    }
    case Action.Type.CHANNEL_MOVE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const from = getChannelIndex(channel, newState.channels);
        if (from >= 0) {
          newState.channels.splice(from, 1);
        }
        const to = findChannelIndex(channel, newState.channels);
        newState.channels.splice(to, 0, channel);
      }
      break;
    }
    case Action.Type.CHANNEL_CLEAR: {
      newState.channels = [];
      break;
    }
  }
  return newState;
};
