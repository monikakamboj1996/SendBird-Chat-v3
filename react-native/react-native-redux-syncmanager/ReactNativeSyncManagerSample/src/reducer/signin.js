
import Action from "../action/signin";

const _initialState = {
	authenticating: false,
	user: null,
	err: null,
	typedUserId: '',
	typedNickname: ''
};

export const reducer = (state = _initialState, action) => {
	const newState = { ...state };
	switch(action.type) {
		case Action.Type.AUTHENTICATE: {
			newState.authenticating = true;
			break;
		}
		case Action.Type.SIGNIN_SUCCESS: {
			newState.authenticating = false;
			newState.user = action.payload.user;
			newState.err = null;
			break;
		}
		case Action.Type.SIGNIN_FAILED: {
			newState.authenticating = false;
			newState.user = null;
			newState.err = action.payload.err;
			break;
		}
	}
	return newState;
};