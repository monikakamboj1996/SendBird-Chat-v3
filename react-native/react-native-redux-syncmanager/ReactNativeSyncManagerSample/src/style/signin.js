
import { StyleSheet } from 'react-native';

export const style = StyleSheet.create({
	container: {
    backgroundColor: '#fff', 
    flex: 1
  },
  logo: {
    marginTop: 35,
    marginBottom: 5,
    alignItems: 'center'
	},
	logoImage: {
		width: 150,
		height: 150
	},
  logoTitleText: {
    color: '#7d62d9', 
    fontSize: 30, 
    fontWeight: '600'
  },
  logoSubtitleText: {
    color: '#7d62d9', 
    fontSize: 13, 
    fontWeight: '500'
  },
  inputContainer: {
    borderWidth: 1,
    borderColor: '#ddd',
    borderRadius: 6,
    backgroundColor:'#fff',
    paddingHorizontal: 8,
    marginTop: 8,
    marginHorizontal: 20
  },
  input: {
    fontSize: 13,
    borderWidth: 0
  },
  buttonContainer: {
    marginTop: 20,
    marginHorizontal: 28
	},
	button: {
    height: 50,
    borderRadius: 6,
		backgroundColor: '#6e5baa'
	},
  errorText: {
    alignSelf: 'center', 
    fontSize: 14, 
    color: '#e03131',
    marginTop: 20
  }
});