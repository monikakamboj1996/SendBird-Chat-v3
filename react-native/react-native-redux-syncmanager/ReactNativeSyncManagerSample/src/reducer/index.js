import { combineReducers } from 'redux';
import { reducer as main } from './main';
import { reducer as chat } from './chat';
import { reducer as invite } from './invite';
import { reducer as member } from './member';
import { reducer as signin } from './signin';

export default combineReducers({
  main,
  chat,
  invite,
  member,
  signin
});
