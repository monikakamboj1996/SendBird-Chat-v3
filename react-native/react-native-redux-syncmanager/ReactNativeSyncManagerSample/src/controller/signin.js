import React from 'react';
import { ActivityIndicator, Image, Text, View } from 'react-native';
import { Button, Input } from 'react-native-elements';
import { connect } from 'react-redux';

import SendBird from 'sendbird';
import SendBirdSyncManager from 'sendbird-syncmanager';

import AsyncStorage from '@react-native-community/async-storage';

import { style } from '../style/signin';
import Action from '../action/signin';

export class SigninController extends React.Component {
  static navigationOptions = {
    header: null
  };
  constructor(props) {
    super(props);
    this.state = {
      authenticating: false,
      user: null,
      err: null,
      typedUserId: '',
      typedNickname: ''
    };
  }
  static getDerivedStateFromProps(props, state) {
    state.authenticating = props.authenticating;
    return state;
  }
  componentDidMount() {
    new SendBirdSyncManager().clearCache();
  }
  connect() {
    const userId = this.state.typedUserId;
    const nickname = this.state.typedNickname;
    if (userId && nickname) {
      const sb = SendBird.getInstance();
      if (sb) {
        this.props.authenticate();
        sb.connect(userId, (_, err) => {
          if (!err) {
            sb.updateCurrentUserInfo(nickname, null, (user, err) => {
              if (!err) {
                AsyncStorage.setItem('currentUser', JSON.stringify(user), () => {
                  this.props.succeed(user);
                  this.props.navigation.replace('Main', { user: user });
                });
              } else {
                this.props.fail(err.message);
              }
            });
          } else {
            this.props.fail(err.message);
          }
        });
      }
    } else {
      this.props.fail('Insert user ID and nickname');
    }
  }
  render() {
    return (
      <View style={style.container}>
        <View style={style.logo}>
          <Image style={style.logoImage} source={require('../img/icon_sb_512.png')} />
          <Text style={style.logoTitleText}>SendBird</Text>
          <Text style={style.logoSubtitleText}>React Native</Text>
        </View>

        <Input
          placeholder="User ID"
          inputContainerStyle={style.inputContainer}
          inputStyle={style.input}
          autoCorrect={false}
          autoCapitalize="none"
          maxLength={16}
          value={this.state.typedUserId}
          onChangeText={text => this.setState({ typedUserId: text })}
          underlineColorAndroid="transparent"
        />

        <Input
          placeholder="Nickname"
          inputContainerStyle={style.inputContainer}
          inputStyle={style.input}
          autoCorrect={false}
          autoCapitalize="none"
          maxLength={16}
          value={this.state.typedNickname}
          onChangeText={text => this.setState({ typedNickname: text })}
          underlineColorAndroid="transparent"
        />

        <View style={style.buttonContainer}>
          <Button
            title="Connect"
            buttonStyle={style.button}
            onPress={this.connect.bind(this)}
            disabled={this.state.authenticating}
          />
        </View>
        <Text style={style.errorText}>{this.props.err || ''}</Text>

        {this.state.authenticating && <ActivityIndicator size="large" color="#6e5baa" />}
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { signin } = previousState;
    return { ...signin };
  },
  Action
)(SigninController);
