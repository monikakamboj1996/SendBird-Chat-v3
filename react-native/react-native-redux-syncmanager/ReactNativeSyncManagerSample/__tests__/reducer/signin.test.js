import { reducer } from '../../src/reducer/signin';
import Action from '../../src/action/signin';

describe('reducer/signin', () => {
  test('checking initial state', () => {
    const action = { type: 'nonexistent' };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  test('AUTHENTICATE', () => {
    const action = {
      type: Action.Type.AUTHENTICATE
    };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  test('SIGNIN_SUCCESS', () => {
    const action = {
      type: Action.Type.SIGNIN_SUCCESS,
      payload: { user: { userId: 1 } }
    };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  test('SIGNIN_FAILED', () => {
    const action = {
      type: Action.Type.SIGNIN_FAILED,
      payload: { err: new Error('authentication failed') }
    };
    expect(reducer(undefined, action)).toMatchSnapshot();
  });
});
