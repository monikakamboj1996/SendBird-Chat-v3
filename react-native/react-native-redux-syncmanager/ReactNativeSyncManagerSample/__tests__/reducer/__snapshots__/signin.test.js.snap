// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer/signin AUTHENTICATE 1`] = `
Object {
  "authenticating": true,
  "err": null,
  "typedNickname": "",
  "typedUserId": "",
  "user": null,
}
`;

exports[`reducer/signin SIGNIN_FAILED 1`] = `
Object {
  "authenticating": false,
  "err": [Error: authentication failed],
  "typedNickname": "",
  "typedUserId": "",
  "user": null,
}
`;

exports[`reducer/signin SIGNIN_SUCCESS 1`] = `
Object {
  "authenticating": false,
  "err": null,
  "typedNickname": "",
  "typedUserId": "",
  "user": Object {
    "userId": 1,
  },
}
`;

exports[`reducer/signin checking initial state 1`] = `
Object {
  "authenticating": false,
  "err": null,
  "typedNickname": "",
  "typedUserId": "",
  "user": null,
}
`;
